Kernel/FileSystem/FileBackedFileSystem.cpp

```diff
@@ -107,115 +107,131 @@ class DiskCache {
     }
 
 private:
-    DiskBackedFS& m_fs;
+    FileBackedFS& m_fs;
     size_t m_entry_count { 10000 };
     KBuffer m_cached_block_data;
     KBuffer m_entries;
     bool m_dirty { false };
 };
 
-DiskBackedFS::DiskBackedFS(BlockDevice& device)
-    : m_device(device)
+FileBackedFS::FileBackedFS(File& file)
+    : m_file_description(FileDescription::create(file))
+{
+}
+FileBackedFS::FileBackedFS(NonnullRefPtr<FileDescription> file_description)
+    : m_file_description(file_description)
 {
 }
 
-DiskBackedFS::~DiskBackedFS()
+FileBackedFS::~FileBackedFS()
 {
 }
 
-bool DiskBackedFS::write_block(unsigned index, const u8* data, FileDescription* description)
+bool FileBackedFS::write_block(unsigned index, const u8* data, FileDescription* description, bool use_logical_block_size)
 {
-#ifdef DBFS_DEBUG
-    klog() << "DiskBackedFileSystem::write_block " << index << ", size=" << data.size();
+    ASSERT(m_logical_block_size);
+#ifdef FBFS_DEBUG
+    klog() << "FileBackedFileSystem::write_block " << index << ", size=" << data.size();
 #endif
 
     bool allow_cache = !description || !description->is_direct();
+    auto selected_block_size = use_logical_block_size ? m_logical_block_size : block_size();
```

Comment:
Naming suggestion: `effective_block_size` (but I'm not sure, English is not my first language, yada yada, all standard disclaimers apply)